spec/conflict_resolver_spec.rb

```diff
@@ -605,7 +605,57 @@ class Z < T::Enum
 
     expect(x.children.length).to be 2
   end
-  
+
+  it 'should deduplicate multiple includes on the same module' do
+    m = gen.root.create_module('M') do |m|
+      m.create_include('I1')
+      m.create_include('I2')
+      m.create_include('I1')
+      m.create_include('I1')
+      m.create_include('I1')
+    end
+
+    expect(m.children.length).to be 5
+
+    subject.resolve_conflicts(m) { |*| raise 'unable to resolve automatically' }
+
+    expect(m.children.length).to be 2
+  end
+
+  it 'should deduplicate multiple extends on the same module' do
+    m = gen.root.create_module('M') do |m|
+      m.create_extend('I1')
+      m.create_extend('I2')
+      m.create_extend('I1')
+      m.create_extend('I1')
+      m.create_extend('I1')
+    end
+
+    expect(m.children.length).to be 5
+
+    subject.resolve_conflicts(m) { |*| raise 'unable to resolve automatically' }
+
+    expect(m.children.length).to be 2
+  end
+
+  it 'should deduplicate multiple extends and multipe includes on the same module' do
+    m = gen.root.create_module('M') do |m|
+      m.create_extend('I1')
+      m.create_extend('I2')
+      m.create_extend('I1')
+      m.create_extend('I1')
+      m.create_include('J1')
+      m.create_include('J1')
+      m.create_include('J1')
+    end
+
+    expect(m.children.length).to be 7
+
+    subject.resolve_conflicts(m) { |*| raise 'unable to resolve automatically' }
+
+    expect(m.children.length).to be 3
+  end
+
```

Comment:
This is missing a test where the same module is included/extended multiple times in a namespace, like so:
```
m = gen.root.create_module('M') do |m|
  m.create_extend('I1')
  m.create_extend('I1')
  m.create_extend('I2')
  m.create_include('I1')
  m.create_include('I1')
  m.create_include('I2')
end
```
and I am afraid with this implementation, this test will fail, since all the children are not equal in this case.